packages/cli/config/tsconfig.json

```diff
@@ -11,7 +11,7 @@
     "incremental": true,
     "isolatedModules": true,
     "jsx": "react",
-    "lib": ["DOM", "DOM.Iterable", "ScriptHost", "ES2020", "ESNext.Promise"],
+    "lib": ["DOM", "DOM.Iterable", "ScriptHost", "ES2020", "ES2020.Promise"],
```

Comment:
I think it's too early to move to `ES2020.Promise`, as `allSettled` isn't supported in all versions of Node.js that we say that we support. Perhaps we could stick to `ES2018.Promise` for now?

This change will also need a changeset for the CLI, given that it's extending the available TypeScript types a `patch` change should be enough.

Reply:
@Rugvip - I think this was added by me a long time ago as part of the CodeOwnersProcessor to use `Promise.any` when looking for the first `CODEOWNERS` file. It's using `core-js` as a polyfill right now - https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/ingestion/processors/codeowners/read.ts#L20.

An alternative is to remove usage of that, but would need to be slightly refactored to use a different library or hand coded. I've also been considering the lookup should maybe be run serially with a priority order based on the orgs preferences in order to avoid using up extra rate limits for files that likely won't exist.

Reply:
Realized this has me confused at a number of levels. `ES2020.Promise` would be included in `ES2020` right? So no need for the separate declaration of that. `Promise.any` also doesn't seem to be part of that, but `ESNext.Promise`, looking here anyway https://github.com/microsoft/TypeScript/blob/master/lib/lib.esnext.promise.d.ts

So if `Promise.any` is not part of these lib declarations, why are type checks passing on this PR? 😅

Reply:
That's a great question. Unless `core-js` is including the typing addition for it.